diff --git a/crates/token_proxy_cli/src/main.rs b/crates/token_proxy_cli/src/main.rs
index 60d8394..90fa7db 100644
--- a/crates/token_proxy_cli/src/main.rs
+++ b/crates/token_proxy_cli/src/main.rs
@@ -114,3 +114,6 @@ async fn serve(paths: token_proxy_core::paths::TokenProxyPaths) -> Result<(), St
     let _ = proxy.stop().await?;
     Ok(())
 }
+
+#[global_allocator]
+static GLOBAL: mimalloc::MiMalloc = mimalloc::MiMalloc;
